locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -867,20 +873,41 @@ protected MachineLocation obtainOnce(ConfigBag setup) throws NoMachinesAvailable
                         String scriptContent = ResourceUtils.create(this).getResourceAsString(setupScriptItem);
                         String script = TemplateProcessor.processTemplateContents(scriptContent, getManagementContext(), substitutions);
                         if (windows) {
-                            ((WinRmMachineLocation)machineLocation).executeCommand(ImmutableList.copyOf((script.replace("\r", "").split("\n"))));
+                            WinRmToolResponse resp = ((WinRmMachineLocation)machineLocation).executeCommand(ImmutableList.copyOf((script.replace("\r", "").split("\n"))));
+                            if (resp.getStatusCode() != 0) {
+                                throw new IllegalStateException("Command 'Customizing node " + this + "' failed with exit code " + resp.getStatusCode() + " for location " + machineLocation);
+                            }
                         } else {
-                            ((SshMachineLocation)machineLocation).execCommands("Customizing node " + this, ImmutableList.of(script));
+                            executeCommandThrowingOnError(
+                                    (SshMachineLocation)machineLocation,
+                                    "Customizing node " + this,
+                                    ImmutableList.of(script));
                         }
                     }
                 }
+                
+                Boolean dontRequireTtyForSudo = setup.get(JcloudsLocationConfig.DONT_REQUIRE_TTY_FOR_SUDO);
+                if (Boolean.TRUE.equals(dontRequireTtyForSudo) ||
+                        dontRequireTtyForSudo == null && setup.get(DONT_CREATE_USER)) {
```

Comment:
Minor: I prefer putting brackets around `dontRequireTtyForSudo == null && setup.get(DONT_CREATE_USER)`, so that folk don't need to worry about the precedence of `||` and `&&`.